src/qt/listctrl.cpp

```diff
@@ -994,41 +1190,99 @@ wxQtListTreeWidget::wxQtListTreeWidget( wxWindow *parent, wxListCtrl *handler )
     m_itemDelegate(handler),
     m_closingEditor(0)
 {
-    connect(this, &QTreeView::clicked, this, &wxQtListTreeWidget::itemClicked);
+    setHeader(new wxQtHeaderView(this));
+
+    setSortingEnabled(true);
+
     connect(this, &QTreeView::pressed, this, &wxQtListTreeWidget::itemPressed);
     connect(this, &QTreeView::activated, this, &wxQtListTreeWidget::itemActivated);
-
-    setItemDelegate(&m_itemDelegate);
-    setEditTriggers(NoEditTriggers);
 }
 
-void wxQtListTreeWidget::EmitListEvent(wxEventType typ,
-                                       const QModelIndex &index) const
+bool wxQtListTreeWidget::EmitListEvent(wxEventType type,
+                                       const QModelIndex &index,
+                                       const ListEventData* data) const
 {
     wxListCtrl *handler = GetHandler();
     if ( handler )
     {
         // prepare the event
         // -----------------
         wxListEvent event;
-        InitListEvent(event, handler, typ, index);
-        EmitEvent(event);
+        InitListEvent(event, handler, type, index);
+
+        if ( !index.isValid() && data )
+        {
+            if ( type == wxEVT_LIST_ITEM_SELECTED ||
+                 type == wxEVT_LIST_ITEM_DESELECTED )
+            {
+                // Instead of sending hundreds of (de)selection messages, send only
+                // one for each range which is more efficient (see issue #4541)
+                // data->first is the first row in the (de)selection
+                // data->second is the last row in the (de)selection
+                wxFAIL_MSG("No implemention yet");
```

Comment:
```suggestion
                wxFAIL_MSG("No implementation yet");
```

But, more importantly, should we really have this assert at all here? This risks being annoying when using the control as a lot of user actions would result in an assert...

Reply:
But no assertion is triggered yet and this is just a reminder for the maintainer if he tries to do a call like this:
```
    const ListEventData data { firstSel, lastSel };
    EmitListEvent(wxEVT_LIST_ITEM_SELECTED, QModelIndex(), &data);
```
which is supposed to send only one `wxEVT_LIST_ITEM_SELECTED` message and the handler can use something like: `event.GetRange();` to get the selected range generating the event (likewise for `wxEVT_LIST_ITEM_DESELECTED`)

but this is the subject of resolving this issue #4541